### ISSUE TITLE
Type casting for envvar config values

### ISSUE THREAD

**Issue by jonathansberry** [ckan/ckan#7593]
## CKAN version
2.10

## Describe the bug
I'm not sure the new config declaration setup is properly casting config values set through environment variables to the declared type. We are setting the CKAN_MAX_UPLOAD_SIZE_MB environment variable, which overrides the `ckan.max_resource_size` config value. It is being imported as a string by CKAN 2.10. It's never cast into an int, and we are therefore getting a TypeError when comparing it's value against the resource size. 

### Steps to reproduce
1. Set the CKAN_MAX_UPLOAD_SIZE_MB environment variable to some value.
2. Try to upload a resource
3. Get an internal server error due to this stack trace:
```
Traceback (most recent call last):
  File "/usr/lib/ckan/ckan/ckan/config/middleware/../../views/api.py", line 283, in action
    result = function(context, request_data)
  File "/usr/lib/ckan/ckan/ckan/logic/__init__.py", line 551, in wrapped
    result = _action(context, data_dict, **kw)
  File "/usr/lib/ckan/ckan/ckan/logic/action/create.py", line 341, in resource_create
    upload.upload(package.resources[-1].id,
  File "/usr/lib/ckan/ckan/ckan/lib/uploader.py", line 335, in upload
    _copy_file(self.upload_file, output_file, max_size)
  File "/usr/lib/ckan/ckan/ckan/lib/uploader.py", line 41, in _copy_file
    if current_size > max_size:
TypeError: '>' not supported between instances of 'int' and 'str'
```

### Expected behavior
The environment variable CKAN_MAX_RESOURCE_SIZE_MB should be imported into the ckan config object as an int not as a string.

### Additional details
A bit of digging seems to reval that the variable max_size (in the stack trace) is a string because it is set using `ckan.lib.uploader.Uploader.get_max_resource_size()`:
https://github.com/ckan/ckan/blob/76ebf1ebab92a9bb08649cfe5e758e91edb616b7/ckan/lib/uploader.py#L95-L96

This simply returns the value of `ckan.max_resource_size` config property. In older versions of CKAN this function also cast the value an int, but this has been removed from v2.10 presumably in favour of the config declaration framework. 

However,  environment variables are imported in `ckan.config.environment.py` using `os.eviron.get()` which I think will always return a string? And whilst I can see a "validation" step (I am somewhat surprised the validation step doesn't check the type), I can't see any step where the value would be cast from the string into the expected type. 

https://github.com/ckan/ckan/blob/76ebf1ebab92a9bb08649cfe5e758e91edb616b7/ckan/config/environment.py#L132-L137

Maybe I have missed something obvious, but I think this is a bug in the CKAN core code?

**Comment by jonathansberry** [comment]
As a temporary work around whilst I await a response to this issue, I am calling the IConfigurable interface from my extension and normalizing the config.  This is definitely not the correct long term solution though.

```
from ckan.common import config_declaration

class MyPlugin(plugins.SingletonPlugin):

    plugins.implements(plugins.IConfigurable)
 
    def configure(config):
        config_declaration.normalize(config)
```

**Comment by smotornyuk** [comment]
Hi, @jonathansberry 

Your solution is pretty good as a hotfix. `IConfigurable.configure` called is not so often, so it won't slow down your application

As for a proper fix, there is a WIP pull request that should fix the problem: #7502. Basically, we are going to move reading envvars before the normalization(at the moment, we are reading them right after normalization, which doesn't have a lot of sense).

**Comment by Vishal-NEC** [comment]
Hi @jonathansberry 
I want to work on this issue.

**Comment by Vishal-NEC** [comment]
@jonathansberry In my opinion Your approach looks fine,
I think there is no effect on the performance after this one liner change in environment.py

**Comment by smotornyuk** [comment]
It's already fixed and will be available since v2.10.2
